OpenVote/src/com/openvote/CastVoteServlet.java

```diff
@@ -1,54 +1,53 @@
 package com.openvote;
 
-
 import java.io.IOException;
-import java.util.Calendar;
-import java.util.Date;
-import java.util.TimeZone;
-import java.util.UUID;
-
+import java.util.*;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-import java.io.IOException;
-
-import javax.servlet.http.*;
+import static com.googlecode.objectify.ObjectifyService.ofy;
  
 
-//COPIED FROM OPENVOTESERVLET.java
-/*
- * key = key.getkey()
- * 
- * */
 @SuppressWarnings("serial")
 public class CastVoteServlet extends HttpServlet {
 
 	public void doGet(HttpServletRequest req, HttpServletResponse resp)
 			throws IOException {
-
-		
 	}
 	
     public void doPost(HttpServletRequest req, HttpServletResponse resp)
             throws IOException {
     	String candidate_str = req.getParameter("candidate");
-    	VoteKey vk = new VoteKey(UUID.randomUUID()); //TODO this needs to be checked for uniqueness
-    	Vote vote = new Vote(vk, Candidate.valueOf(candidate_str));
     	
+    	// cast real vote
+    	Vote vote = new Vote(Candidate.valueOf(candidate_str).ordinal());
+    	ofy().save().entity(vote).now(); 	
+    	//TODO: verify that the autogenerated id is unique
     	
-		resp.setContentType("text/plain");
-		resp.getWriter().println("You voted for " + candidate_str + ".");
-		resp.getWriter().println("Your key is " + vote.getVoteKey().toString() + ".");
-		
+    	// cast vote for all fake candidates
+    	ArrayList<Vote> fakeVotes = new ArrayList<Vote>();
+		for (Candidate c: Candidate.values()) {
+			Vote fakeVote = new Vote(c.ordinal());
+    		ofy().save().entity(fakeVote).now();
+    		fakeVotes.add(fakeVote);
+    	}
+    	
+    	// pass real and fake vote information to display keys page
+		req.setAttribute("realVote", vote);
```

Comment:
We can leave this out, and when we want to display things to people just tell them "please check that your indended candidate has one more vote than all others."